Apply co-break-word within resource lists, to Name & Namespace columns selectively so that long strings wrap appropriately.

sg00dwin opened the pull request:
Related fix https://jira.coreos.com/browse/CONSOLE-560

sg00dwin commented:
/retest

spadgett commented:
I'd prefer `Overflow` to `OverflowYFade` in most cases since you can select the text.

benjaminapetersen commented on frontend/public/components/secscan/pod-vuln.jsx:
```diff
@@ -120,18 +120,16 @@ const ContainerLink = ({podvuln, name}) => {
-      <div className="col-md-2 hidden-sm hidden-xs">
-        <ContainerLink podvuln={podvuln} name={imgvuln.Container} />
-      </div>
+    <div className="col-sm-2 col-xs-4">
```
This is a slightly different issue, correct?  Looks like you removed markup instead of adding `<OverflowYFade>`.

benjaminapetersen commented on frontend/public/components/_resource.scss:
```diff
@@ -95,3 +95,7 @@ $height: 18px;
 .co-m-resource-icon--align-left {
   margin-left: 0;
 }
+
```
I don't see this class used anywhere.

sg00dwin replied:
`.co-m-resource-icon--align-left` is used on the ingress view.

benjaminapetersen commented on frontend/public/components/_list.scss:
```diff
@@ -13,11 +13,6 @@
   }
 }
 
```
+1 for removing, `middler` is a very vague classname. :)

sg00dwin replied:
`middler` looked to be used randomly and set `display:flex; align-items: center`
`.co-resource-list-item` now controls that.

benjaminapetersen commented on frontend/public/components/image-stream.tsx:
```diff
@@ -4,7 +4,7 @@ import * as _ from 'lodash-es';
 // eslint-disable-next-line no-unused-vars
 import { K8sResourceKindReference } from '../module/k8s';
 import { ColHead, DetailsPage, List, ListHeader, ListPage } from './factory';
```
@spadgett mentioned `<Overflow>` vs `<OverflowYFade>`.  I'm not sure what the other component does, but the screenshots look good (I didn't pull & run this branch).

sg00dwin replied:
Both `Overflow` and `OverflowYFade` visually do the same, but `Overflow` adds an onClick handler that selects the text.

benjaminapetersen replied:
Wouldn't have guessed that by the names :)

sg00dwin commented:
Updates to pr 
- Switched `< OverflowYFade>` to `< Overflow>` in cases where text are non links so that it adds an onClick handler that selects the text.
- moved style rules `display: flex` and `align-items: center` from `.row` to `.co-resource-list__item` so that `.row` can possibly removed in a later pr

sg00dwin commented:
/retest

spadgett commented on frontend/public/components/RBAC/_rbac.scss:
```diff
@@ -3,6 +3,7 @@
 }
 
 .rbac-rule {
+  width: 100%; // since it's parent .co-resource-list__item is a flex container
```
nit: should be "its" with no apostrophe :)

benjaminapetersen replied:
Rather than a comment, can we encode that?  

```CSS
.co-resource-list__item {
   .rbac-rule {
      width: 100% 
   }
}
```
BEM doesn't approve of the nesting, but the comment reveals these are coupled via nesting anyway.  Perhaps both are helpful?

spadgett commented on frontend/public/components/_list.scss:
```diff
@@ -29,14 +24,13 @@
     margin-left: 10px;
   }
-  [class^="col-"] {
-    @include co-break-word;
```
Yeah, this seemed a bit heavy handed. Glad to see it more selectively applied

spadgett commented on frontend/public/components/build-config.tsx:
```diff
@@ -89,13 +89,13 @@ const BuildConfigsHeader = props => <ListHeader>
 
 const BuildConfigsRow: React.SFC<BuildConfigsRowProps> = ({obj}) => <div className="row co-resource-list__item">
-  <div className="col-xs-3">
+  <OverflowYFade className="col-xs-3">
```
I wonder if we shouldn't let names wrap. They can be important.

spadgett replied:
Minimally we need to make sure there's at least a `title` attribute for `Overflow` / `OverflowYFade` so you can see the full name on hover.

spadgett commented:
jenkins push

coreos-ui commented:
The PR image failed.

spadgett commented:
I don't think we can use this for name because many resource names start with the same prefix, and the identifying part might be hidden. For example, there's no way to tell these pods from the same deployment apart:

<img width="310" alt="screen shot 2018-06-21 at 7 49 03 pm" src="https://user-images.githubusercontent.com/1167259/41750868-520321a2-758c-11e8-8481-0774f3544b1e.png">

/hold

sg00dwin commented:
@spadgett I removed the OverflowYFade from resource names and made them so they'd wrap instead.

spadgett commented:
@rhamilto fyi

sg00dwin commented:
/retest

rhamilto commented:
Seems reasonable to me.  I'm not a huge fan of `<OverflowYFade>` and wonder if it impacts accessibility?

sg00dwin commented:
added an `aria-label` value to the `input` for accessibility.

sg00dwin commented:
@spadgett PTAL

sg00dwin commented:
/retest

spadgett commented:
yarn run lint

sg00dwin commented:
/retest

sg00dwin commented:
@spadgett 
fixed the lint issue

spadgett commented on frontend/public/components/routes.tsx:
```diff
@@ -140,12 +140,12 @@ const RouteListHeader: React.SFC<RouteHeaderProps> = props => <ListHeader>
       namespace={route.metadata.namespace} title={route.metadata.uid} />
   </div>
-  <div className="col-lg-2 col-md-3 col-sm-4 col-xs-6">
+  <div className="col-lg-2 col-md-3 col-sm-4 col-xs-6 co-break-word">
```
co-break-word on route location as well?

spadgett commented:
/retest
